model/riscv_vmem.sail

```diff
@@ -164,7 +164,30 @@ function pt_walk(sv_params,
                 }
               }
               else {
+                let mask_bits = level * sv_params.pte_PPN_j_size_bits;
+                let mask : bits(64) = ~ (ones() << mask_bits);
+                let ppn : bits(64) = ppns | (vpns_of_va(sv_params, va) & mask);
+                let napot_mask : bits(64) = ppn ^ (ppn - 1);
+                let napot_mask_not : bits(64) = ~ (napot_mask);
+                let ptshift_level = sv_params.levels - 1;
+                let ptshift = ptshift_level * sv_params.vpn_size_bits;
+
+                let page_base = (ppn & napot_mask_not) | 
+                                (vpns_of_va(sv_params, va) & napot_mask) | 
+                                (vpns_of_va(sv_params, va) & ((zero_extend(0b1) << ptshift) - zero_extend(0b1)));
+
                 let pa = (ppns << pagesize_bits) | zero_extend(offset_of_va(va));
+                let pa_napot = page_base << 12 | zero_extend(offset_of_va(va));
+
+                // For svnapot ext purpose.
+                if (pte & (zero_extend(0b1) << 63)) == (zero_extend(0b1) << 63) then {
+                  if (napot_mask != zero_extend(0xf)) then {
+                    return PTW_Failure(PTW_Invalid_PTE(), ext_ptw)
```

Comment:
Did you mean `PTW_Unsupported_Napot` here?

I'd probably call it `PTW_Reserved_Napot` btw.